tools-core/src/main/java/com/hedera/hashgraph/client/core/json/Identifier.java

```diff
@@ -147,6 +148,14 @@ public static Identifier parse(final String id) {
 			throw new HederaClientRuntimeException("The provided string was null or empty");
 		}
 
+		if (id.contains("(")) {
```

Comment:
nit: maybe add a comment to explain what's the expected string pattern for the `id`. This will help to understand why it's necessary to have a recursive algorithm to parse the id string.

Reply:
Good idea I will add a comment detailing the possible acceptable string